crates/astria-optimism/src/contract.rs

```diff
@@ -0,0 +1,102 @@
+//! This module contains functions for interacting with the OptimismPortal.sol contract,
+//! which is part of Optimism Bedrock's L1 contracts.
+//! See [the contract](https://github.com/ethereum-optimism/optimism/blob/9a13504bb1f302ca9d412589aac18d589c055f16/packages/contracts-bedrock/src/L1/OptimismPortal.sol).
+//!
+//! This function requires an `OptimismPortal.abi` file in the project root, which
+//! contains the ABI of the above contract. This is used for generating the Rust
+//! contract bindings with abigen.
+use std::sync::Arc;
+
+use ethers::prelude::*;
+use eyre::WrapErr as _;
+use k256::ecdsa::SigningKey;
+
+abigen!(
+    OptimismPortal,
+    "./OptimismPortal.abi",
+    methods {
+        l2Oracle() as renamedl2Oracle;
+    },
+);
+
+/// Returns a new read-only [`OptimismPortal`] contract instance.
+pub fn get_optimism_portal_read_only<P: JsonRpcClient>(
```

Comment:
As noted in other cases, I like to call these `make_` or `create_` or `construct_` insteaed of `get_`, because `get_` feels like you are getting a reference to an object initialized at some `static` memory location.